### DISCUSSION TITLE
Parameters behavior

### DISCUSSION THREAD

**Discussion by mtovts** [OAI/OpenAPI-Specification#2654] [Q&A] [answered]
Hello!
Specification [says](https://github.com/OAI/OpenAPI-Specification/blob/d9ac75b00c8bf405c2c90cfa9f20370564371dec/versions/3.0.3.md#parameter-object) `A unique parameter is defined by a combination of a name and location.`

In [`PathItem`](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#path-item-object) specification says `PathItem parameters can be overridden at the operation level, but cannot be removed there.`

In [`Operation`](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#operation-object) specification says `If a parameter is already defined at the PathItem, the new definition will override it but can never remove it.`

I defined following OpenAPI
```yaml
openapi: 3.0.3
info:
  title: Parameters example
  version: 0.1.0
servers:
  - url: 'https://127.0.0.1/api/rest'
paths:
  '/foo/{bar}':
    post:
      parameters:
        - name: bar
          in: header
          schema:
            type: string
      requestBody:
        content:
          application/json:
            schema:
              type: string
      responses:
        '200':
          description: Example
          content:
            application/json:
              schema:
                type: string
    delete:
      responses:
        '200':
          description: OK
    parameters:
      - required: true
        name: bar
        in: path
        schema:
          type: string
```
I defined `foo` parameter in path at `PathItem` and I defined parameter with same name `foo`, in header at `Operation`. I expect two parameters in my post. But `swaggerhub` UI says that parameter `foo` is overriden in my post and I've got only in header `foo` parameter. How it should work?

**Comment by hkosova** [comment, marked as answer]
Your API definition is correct - it's OK to have parameters with the same `name` as long as they have different location (i.e. path vs header).

The issue you're seeing is a bug in Swagger UI. Please submit a bug report here:
https://github.com/swagger-api/swagger-ui/issues

**Reply by mtovts** [reply]
Thank you!
I've created the [issue](https://github.com/swagger-api/swagger-ui/issues/7430).

**Comment by handrews** [comment]
Resolved by filing an issue elsewhere, so closing here.
